Compiler/src/abstractinterpretation.jl

```diff
@@ -3485,7 +3485,39 @@ function refine_partial_type(@nospecialize t)
     return t
 end
 
+function abstract_eval_nonlinearized_foreigncall_name(interp::AbstractInterpreter, e, sstate::StatementState, sv::AbsIntState)
+    if isexpr(e, :call)
+        n = length(e.args)
+        argtypes = Vector{Any}(undef, n)
+        for i = 1:n
+            argtypes[i] = abstract_eval_nonlinearized_foreigncall_name(interp, e.args[i], sstate, sv)
+        end
+        result = abstract_call(interp, ArgInfo(e.args, argtypes), sstate, sv)
+    else
+        result = abstract_eval_basic_statement(interp, e, sstate, sv)
+    end
+    if result isa Future
+        if !isready(result)
+            result
```

Comment:
I think things can break if we hit this case, so possibly we should bail out if this happens. Need some advice on this.

Reply:
`abstract_eval_statement_expr` which calls `abstract_eval_foreigncall` returns a future, so you can just wrap the continuation in abstract_eval_foreigncall even if you're only using the future for its effects.